pkg/logs/sender/sender.go

```diff
@@ -6,220 +6,125 @@
 package sender
 
 import (
-	"context"
+	"time"
 
 	"github.com/DataDog/datadog-agent/pkg/logs/client"
 	"github.com/DataDog/datadog-agent/pkg/logs/message"
-	"github.com/DataDog/datadog-agent/pkg/logs/metrics"
 )
 
-// Strategy should contain all logic to send logs to a remote destination
-// and forward them the next stage of the pipeline.
-type Strategy interface {
-	Send(inputChan chan *message.Message, outputChan chan *message.Message, send func([]byte) error)
-	Flush(ctx context.Context)
+type destinationState struct {
+	input       chan *message.Payload
+	destination client.Destination
+	stopChan    chan struct{}
 }
 
-// Sender contains all the logic to manage a stream of messages to destinations
-type Sender interface {
-	Start()
-	Stop()
-	Flush(ctx context.Context)
-}
-
-// SingleSender sends logs to different destinations.
-type SingleSender struct {
-	inputChan    chan *message.Message
-	outputChan   chan *message.Message
-	hasError     chan bool
+// Sender sends logs to different destinations.
```

Comment:
This would be a good spot to summarize the intended behavior with respect to reliable and additional destinations.  Some comments below detail how I'm not sure this behavior is implemented correctly, and having a nearby reference for the expected behavior would make it much easier to reason about it, both now and when this code is later modified.

Reply:
I think this comment is still valid ^

Reply:
Missed this one thanks!